### PR TITLE
fix(schema): Don't report changed keys as deprecated

### PR DESCRIPTION
**Pull request by holmanb**
## Proposed Commit Message
```
fix(schema): Don't report changed keys as deprecated
```

## Additional Context
also fixes issue where unittests run with version boundary set

### CODE REVIEW

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Comment by holmanb** [comment]
updated the commit messages

File: cloudinit/config/schema.py [hunk 1]
```diff
@@ -797,8 +797,9 @@ def validate_cloudconfig_schema(
         ):  # pylint: disable=W1116
             if (
                 "devel" != features.DEPRECATION_INFO_BOUNDARY
-                and Version.from_str(schema_error.version)
-                > Version.from_str(features.DEPRECATION_INFO_BOUNDARY)
+                and (schema_error.version == "devel"
+                or Version.from_str(schema_error.version)
+                > Version.from_str(features.DEPRECATION_INFO_BOUNDARY))
```
Comment (blackboxsw) [inline cloudinit/config/schema.py:802]: this won't work for our default "devel" case as Version.from_str("devel") will traceback
```
ValueError: invalid lteral for int() with base 10: 'devel'
```

Comment (blackboxsw) [reply cloudinit/config/schema.py:802]: Resolving this comment as we only see schema_error.version == "devel" when no deprecated_version is set in the schema definition, which happens to be when we have the changed_version changed_message keys... Which I realize is what @holmanb was proposing to fix here.

I also can confirm this fix doesn't report deprecation messages for "changed" keys

**Pull request merged**